### ISSUE TITLE
getMany() works but getManyAndCount() doesn't and throws error

### ISSUE THREAD

**Issue by HRK44** [typeorm/typeorm#9299]
## Issue Description
Here's my simple query : 
```javascript
        const query = Toto.createQueryBuilder('toto')
            .select(['toto'])
            .leftJoinAndSelect('toto.tag', 'tag')
            .leftJoinAndSelect('tag.data', 'tagData')
            .where({ foo: 'bar' })
            .skip(0)
            .take(25);

await query.getMany(); //works;

//QueryFailedError: invalid reference to FROM-clause entry for table "tag"
await query.getManyAndCount(); //doesnt work


class Toto {

    @ManyToOne(() => Tag, { nullable: true })
    @JoinColumn()
    tag?: Tag;

}

class Tag {
    @OneToMany(() => TagData, tagData => tagData.tag, {
        cascade: true,
        onDelete: 'CASCADE',
        onUpdate: 'CASCADE'
    })
    data: TagData[];
}
```

`QueryFailedError: invalid reference to FROM-clause entry for table "tag"`

Looks like this join makes it blow up `.leftJoinAndSelect('tag.data', 'tagData')`

### Expected Behavior
Get the count

### Actual Behavior
Error:  QueryFailedError: invalid reference to FROM-clause entry for table "tag"

### My Environment

typeorm / nodejs

**Comment by HRK44** [comment]
Actually the issue seems to happen with some unrelated leftJoinAndSelect, once I have like 3 joins/select it stops working

**Comment by HRK44** [comment]
Fixed after upgrading to 0.2.45
